src/core/toga/widgets/table.py

```diff
@@ -44,6 +44,7 @@ def __init__(self, headings, id=None, style=None, data=None, accessors=None, on_
         self.headings = headings
         self._accessors = build_accessors(headings, accessors)
 
+        self._on_select = None
```

Comment:
Yes. This primes the attribute value so that `on_select` property requests *always* work. This avoids the need to do `hasattr(interface, 'on_select')` later on.